src/consumption.cpp

```diff
@@ -215,191 +227,157 @@ morale_type player::allergy_type( const item &food ) const
     return MORALE_NULL;
 }
 
-bool player::is_allergic( const item &food ) const
+edible_rating player::can_eat( const item &food, std::string *err ) const
```

Comment:
> The interface would be cleaner if the return value included the error - return by parameter is C-ish.

Agreed. For now, I've simply repeated the signature of player::can_eat

> For example, if you returned it as a const std::pair<edible_rating, std::string> &

I was thinking about that too and came up with a template class. Something like this:

```c++
template<typename T = bool, T no_error = true, T default_error = false>
class ret_val
{
    public:
        static_assert( no_error != default_error, "Error codes must differ." );

        static ret_val success( const std::string &msg = std::string() ) {
            return ret_val( no_error, msg );
        }

        static ret_val failure( T code = default_error, const std::string &msg = std::string() ) {
            return ret_val( default_error, msg );
        }

        operator bool() const {
            return code == no_error;
        }

        T operator *() const {
            return code;
        }

        const std::string &get_msg() const {
            return msg;
        }

    private:
        ret_val( T code, const std::string &msg ) : code( code ), msg( msg ) {}

    private:
        std::string msg;
        T code;
};

```
I think this should be done in a separate PR for all related cases.